packages/indexer-common/src/indexer-management/actions.ts

```diff
@@ -12,13 +15,61 @@ import { Eventual, join, Logger, timer } from '@graphprotocol/common-ts'
 export class ActionManager {
   constructor(
     public allocationManager: AllocationManager,
+    public networkMonitor: NetworkMonitor,
+    private currentEpoch: number,
+    private maxAllocationEpoch: number,
     private logger: Logger,
     private models: IndexerManagementModels,
+    private allocationManagementMode?: AllocationManagementMode,
+    private autoMinBatchSize?: number,
+    private autoMaxWaitTime?: number,
   ) {}
 
+  private async batchReady(
+    approvedActions: Action[],
+    pollingFrequency: number,
+  ): Promise<boolean> {
+    if (approvedActions.length < 1) {
+      return false
+    }
+
+    // In AUTO mode, worker waits to execute when 1) the batch is not filled upto minimum size
+    // and 2) oldest allocatation is not expiring after the current epoch
+    // and 3) oldest action is not stale (10 times the user provided threshold of polling frequency)
+    if (this.allocationManagementMode === AllocationManagementMode.AUTO) {
+      const oldestCreatedAtEpoch = (
+        await this.networkMonitor.allocations(AllocationStatus.ACTIVE)
+      )[0].createdAtEpoch
+
+      const updatedDeadline = new Date(new Date().valueOf() - 10 * pollingFrequency)
+
+      this.logger.debug('Worker in AUTO, checking batch conditions', {
```

Comment:
There's an info log `Executing batch of approved actions` for when `batchReady` is true, and the conditional logging at line92 is the complement of that. I guess they were mainly for me to check the results quickly, but I could remove both logs if `Executing batch of approved actions` is enough?

Reply:
Yeah, I think `Executing batch of approved actions` is enough as long as it includes all the info. Do you think having only that would make it unclear what's happening in the logs?

Reply:
I think it might be tricky to include all the info in `Executing batch of approved actions` as it lives outside of `batchReady`, thinking about this again, I'll add a log after conditional tests like you suggested that shows the results:)

Reply:
ok!